File: samps/src/ipractice.rb
#!/bin/env ruby
#
# File:    ipractice.rb
# Purpose: sum of positive integers practice problem
# Author:  pc2@ecs.csus.edu
#


line = gets
inval = line.to_i

until 0 == inval
  line = gets

  sum = 0

  if inval < 1
     1.downto(inval) do |i|
       sum = sum + i
     end
  else
     1.upto(inval).each do |i|
       # puts("#{inval} and #{i} sum #{sum}")
       sum = sum + i
     end
  end

  printf "N = %-3d    Sum = %d\n", inval, sum
  inval = line.to_i

end 

# eof ipractice.rb

File: samps/src/ipractice.cs
/*
 *
 * File:    isumit.cs
 * Purpose: to sum the positive from stdin
 * pc2@ecs.csus.edu at http://www.ecs.csus.edu/pc2
 *
 * $Id: hello.cs 3159 2015-09-22 20:39:11Z boudreat $
*/
using System;
 
public class iSumit
{
    static public void Main ()
    {
        int sum = 0;
        string str;
        while((str = Console.ReadLine()) != null) 
        {
            int intval = int.Parse(str);
            int i;

          if (inval < 1)
          {
            for (i = 1; i >= inval; i --)
                   sum += i;
          }
          else
          {
            for (i = 1; i <= inval; i ++)
                   sum += i;
          }

          Console.WriteLine ("The sum of the integers is {0}",sum);
        }
    }
}
